src/app/helpers/page/pageComponent.ts

```diff
@@ -5,44 +5,34 @@ import { withUnsubscribe } from "../unsubscribe/unsubscribe";
 import { IPageInfo, PageInfo } from "./pageInfo";
 
 export interface IPageComponent {
-  readonly pageInfo: PageInfo;
+  readonly pageInfos: PageInfo[];
 }
 
 export interface IPageComponentStatic extends Type<IPageComponent> {
-  readonly pageInfo: PageInfo;
+  readonly pageInfos: PageInfo[];
 }
 
 export class PageComponent extends withUnsubscribe() implements IPageComponent {
-  private static _pageInfo: PageInfo;
+  private static _pageInfos: PageInfo[];
 
-  public static get pageInfo(): PageInfo {
-    return this._pageInfo as PageInfo;
+  public static get pageInfos(): PageInfo[] {
+    return this._pageInfos;
   }
 
-  public get pageInfo() {
-    return PageComponent.pageInfo;
+  public get pageInfos(): PageInfo[] {
+    return PageComponent.pageInfos;
   }
 
   /**
    * Creates and links the page info for the page component
    */
-  public static linkComponentToPageInfo(info: IPageInfo): typeof PageComponent {
-    this._pageInfo = new PageInfo(info);
-    return this;
-  }
-
-  /**
-   * Sets the menu route for the page component, and updates it to link to the
-   * page info
-   */
-  public static andMenuRoute(menu: MenuRoute): typeof PageComponent {
-    if (!this._pageInfo) {
-      throw new Error(
-        "AndMenuRoute must be called after LinkComponentToPageInfo"
-      );
-    }
-
-    this._pageInfo.setMenuRoute(this, menu);
+  public static linkToRouterWith(
+    info: IPageInfo,
+    menu: MenuRoute
+  ): typeof PageComponent {
```

Comment:
Naming discussion: would this be better named `linkToRoute` - because that's really what's happening here, we are telling the router which component to render for a route.

Also: might be worth considering if we need the entire menu item in this interface? I don't know, but in the interests on minimalism it would be great if we only accepted a strongRoute here over a `menuRoute`.

Actually, on looking at the `pageInfo.SetMenuRoute` call below, why are we even accepting another argument. Could we just pass the `pageInfo` to this method since pageInfo can store a route?

There might have been a historical reason (probably the way decorators worked) that we needed to pass the route separately from the pageinfo, but I can't think of a good reason now?

Reply:
Looking through the code, we do actually use the information stored in the menu route for the secondary menu. Otherwise the other changes should be feasible and i will attempt to do so